src/components/ErrorDisplay.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+interface ErrorDisplayProps {
+  errors: Record<string, { message?: string }> | undefined
+  fieldName: string
+}
```

Comment:
Why not simply:
```
interface ErrorDisplayProps {
  errorMessage: string | undefined
}
```
And then you use it with:
```
<ErrorDisplay errorMessage={errors.username} />
```
Also maybe we want to call that component something like FormFieldError or something that better describes it

Reply:
eslint doesn't let me make it simple he needs it to be more explicit

Reply:
My point is why do we need 2 fields when it can be one being the actual error directly.
Adjust my type suggestion to make eslint happy, idea is the same

Reply:
done